src/plugins/saved_objects_management/server/routes/scroll_count.ts

```diff
@@ -39,34 +39,65 @@ export const registerScrollForCountRoute = (router: IRouter) => {
       validate: {
         body: schema.object({
           typesToInclude: schema.arrayOf(schema.string()),
+          namespacesToInclude: schema.maybe(schema.arrayOf(schema.string())),
           searchString: schema.maybe(schema.string()),
         }),
       },
     },
     router.handleLegacyErrors(async (context, req, res) => {
       const { client } = context.core.savedObjects;
+      const namespaces = [];
+      if (req.body.namespacesToInclude && req.body.namespacesToInclude.length > 0) {
+        req.body.namespacesToInclude.forEach((ns) => {
+          if (ns === null) {
+            namespaces.push('default');
+          } else {
+            namespaces.push(ns);
+          }
+        });
+      }
 
       const findOptions: SavedObjectsFindOptions = {
+        ...req.body.params,
         type: req.body.typesToInclude,
         perPage: 1000,
       };
+
+      if (namespaces.length > 0) {
+        findOptions.namespaces = namespaces;
+      }
+
       if (req.body.searchString) {
         findOptions.search = `${req.body.searchString}*`;
         findOptions.searchFields = ['title'];
       }
 
       const objects = await findAll(client, findOptions);
 
-      const counts = objects.reduce((accum, result) => {
+      const counts = {
+        type: {},
+        namespaces: {},
+      };
+
+      objects.forEach((result) => {
         const type = result.type;
-        accum[type] = accum[type] || 0;
-        accum[type]++;
-        return accum;
-      }, {} as Record<string, number>);
+        if (req.body.namespacesToInclude) {
+          const resultNamespaces = (result.namespaces || []).flat();
+          resultNamespaces.forEach((ns) => {
+            if (ns === null) {
+              ns = 'default';
+            }
+            counts.namespaces[ns] = counts.namespaces[ns] || 0;
+            counts.namespaces[ns]++;
+          });
+        }
+        counts.type[type] = counts.type[type] || 0;
+        counts.type[type]++;
+      });
 
       for (const type of req.body.typesToInclude) {
-        if (!counts[type]) {
-          counts[type] = 0;
+        if (!counts.type[type]) {
+          counts.type[type] = 0;
         }
```

Comment:
dont we want something similar for namespaces where we have a default count of 0 for all `namespacesToInclude`?

Reply:
Added, it didn't cause any errors because there is logic on the filterOption creation to display 0 if its not in the counts data structure, but it makes sense for this API to return 0 for namespaces that are explicitly requested instead of not having the namespace in the response.